modules/data/src/entity-data.module.ts

```diff
@@ -2,60 +2,25 @@ import { ModuleWithProviders, NgModule } from '@angular/core';
 
 import { EffectsModule, EffectSources } from '@ngrx/effects';
 
-import { DefaultDataServiceFactory } from './dataservices/default-data.service';
-
-import {
-  DefaultPersistenceResultHandler,
-  PersistenceResultHandler,
-} from './dataservices/persistence-result-handler.service';
-
-import {
-  DefaultHttpUrlGenerator,
-  HttpUrlGenerator,
-} from './dataservices/http-url-generator';
-
-import { EntityCacheDataService } from './dataservices/entity-cache-data.service';
 import { EntityCacheEffects } from './effects/entity-cache-effects';
-import { EntityDataService } from './dataservices/entity-data.service';
 import { EntityEffects } from './effects/entity-effects';
 
-import { ENTITY_METADATA_TOKEN } from './entity-metadata/entity-metadata';
-
-import {
-  ENTITY_CACHE_META_REDUCERS,
-  ENTITY_COLLECTION_META_REDUCERS,
-} from './reducers/constants';
-import { Pluralizer, PLURAL_NAMES_TOKEN } from './utils/interfaces';
-import { DefaultPluralizer } from './utils/default-pluralizer';
-
+import { EntityDataModuleConfig } from './entity-data-config';
 import {
-  EntityDataModuleConfig,
-  EntityDataModuleWithoutEffects,
-} from './entity-data-without-effects.module';
+  ENTITY_DATA_PROVIDERS,
+  ENTITY_DATA_WITHOUT_EFFECTS_PROVIDERS,
+  initializeEntityData,
+  provideRootEntityData,
+  provideRootEntityDataWithoutEffects,
+} from './provide-entity-data';
 
 /**
  * entity-data main module includes effects and HTTP data services
  * Configure with `forRoot`.
  * No `forFeature` yet.
  */
 @NgModule({
-  imports: [
-    EntityDataModuleWithoutEffects,
-    EffectsModule, // do not supply effects because can't replace later
-  ],
-  providers: [
-    DefaultDataServiceFactory,
-    EntityCacheDataService,
-    EntityDataService,
-    EntityCacheEffects,
-    EntityEffects,
-    { provide: HttpUrlGenerator, useClass: DefaultHttpUrlGenerator },
-    {
-      provide: PersistenceResultHandler,
-      useClass: DefaultPersistenceResultHandler,
-    },
-    { provide: Pluralizer, useClass: DefaultPluralizer },
-  ],
+  providers: [ENTITY_DATA_WITHOUT_EFFECTS_PROVIDERS, ENTITY_DATA_PROVIDERS],
```

Comment:
```suggestion
  imports: [EntityDataModuleWithoutEffects],
  providers: [ENTITY_DATA_PROVIDERS],
```

btw, there are unused imports in this file (EffectsModule, EntityEffects, etc.) that should be removed